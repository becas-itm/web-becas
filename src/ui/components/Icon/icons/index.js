export { ArrowBack } from './ArrowBack';

export { ArrowForward } from './ArrowForward';

export { Close } from './Close';

export { Event } from './Event';

export { Money } from './Money';

export { Search } from './Search';

export { Tune } from './Tune';
