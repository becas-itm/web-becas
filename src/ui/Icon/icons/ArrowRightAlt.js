import React from 'react';
import Icon from '../Icon';

export function ArrowRightAlt(props) {
  return (
    <Icon {...props} viewBox="0 0 24 24">
      <path d="M0 0h24v24H0z" fill="none" />
      <path d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z" />
    </Icon>
  );
}

ArrowRightAlt.defaultProps = Icon.defaultProps;

ArrowRightAlt.propTypes = Icon.propTypes;
