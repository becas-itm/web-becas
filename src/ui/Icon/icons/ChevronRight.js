import React from 'react';
import Icon from '../Icon';

export function ChevronRight(props) {
  return (
    <Icon {...props} viewBox="0 0 24 24">
      <path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" />
      <path d="M0 0h24v24H0z" fill="none" />
    </Icon>
  );
}

ChevronRight.defaultProps = Icon.defaultProps;

ChevronRight.propTypes = Icon.propTypes;
