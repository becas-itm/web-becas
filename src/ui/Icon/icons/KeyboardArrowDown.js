import React from 'react';
import Icon from '../Icon';

export function KeyboardArrowDown(props) {
  return (
    <Icon {...props} viewBox="0 0 24 24">
      <path d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z" />
      <path d="M0 0h24v24H0V0z" fill="none" />
    </Icon>
  );
}

KeyboardArrowDown.defaultProps = Icon.defaultProps;

KeyboardArrowDown.propTypes = Icon.propTypes;
