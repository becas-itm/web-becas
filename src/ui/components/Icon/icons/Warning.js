import React from 'react';
import Icon from '../Icon';

export function Warning(props) {
  return (
    <Icon {...props} viewBox="0 0 24 24">
      <path d="M0 0h24v24H0z" fill="none" />
      <path d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z" />
    </Icon>
  );
}

Warning.defaultProps = Icon.defaultProps;

Warning.propTypes = Icon.propTypes;
