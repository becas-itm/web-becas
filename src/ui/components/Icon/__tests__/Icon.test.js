import React from 'react';
import { render } from '@testing-library/react';

import * as icons from '../icons';
import { Icon, SIZE } from '../index';

const Content = () => <div data-testid="icon-content" />;

const TestIcon = props => (
  <Icon data-testid="icon" {...props}>
    <Content />
  </Icon>
);

describe('Icon component', () => {
  test('renders correctly', () => {
    const { getByTestId } = render(<TestIcon />);
    expect(getByTestId('icon-content')).toBeInTheDocument();
  });

  describe('icon size', () => {
    it('should be square', () => {
      const { getByTestId } = render(<TestIcon />);
      expect(getByTestId('icon')).toHaveClass('-sizeRegular');
    });

    it('size auto should fill the available space', () => {
      const sizeAuto = SIZE.auto;
      const { getByTestId } = render(<TestIcon size={sizeAuto} />);
      expect(getByTestId('icon')).toHaveClass('-sizeAuto');
    });

    it('size can be small', () => {
      const sizeSmall = SIZE.small;
      const { getByTestId } = render(<TestIcon size={sizeSmall} />);
      expect(getByTestId('icon')).toHaveClass('-sizeSmall');
    });
  });

  describe('icon title', () => {
    it('null title should have aria-hidden attribute', () => {
      const { getByTestId } = render(<TestIcon />);
      expect(getByTestId('icon')).toHaveAttribute('aria-hidden', 'true');
    });

    it('role attribute should be `img`', () => {
      const title = 'foo';
      const { getByTestId } = render(<TestIcon title={title} />);
      const icon = getByTestId('icon');

      expect(icon).toHaveAttribute('role', 'img');
      expect(icon).not.toHaveAttribute('aria-hidden');
    });

    it('should render a title element', () => {
      const title = 'foo';
      const { getByTitle } = render(<TestIcon title={title} />);
      expect(getByTitle(title)).toHaveTextContent(title);
    });
  });

  describe('all icons', () => {
    test.each(Object.entries(icons))(
      '%s icon renders correctly',
      (icon, IconComp) => {
        const { getByTestId } = render(<IconComp data-testid={icon} />);
        expect(getByTestId(icon)).toMatchSnapshot();
      },
    );
  });
});
