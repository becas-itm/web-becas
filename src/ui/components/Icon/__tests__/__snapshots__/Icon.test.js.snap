// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component all icons ArrowBack icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="ArrowBack"
  viewBox="0 0 24 24"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
  />
</svg>
`;

exports[`Icon component all icons ArrowForward icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="ArrowForward"
  viewBox="0 0 24 24"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
  />
</svg>
`;

exports[`Icon component all icons Close icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="Close"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon component all icons Event icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="Event"
  viewBox="0 0 24 24"
>
  <path
    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19a2 2 0 002 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon component all icons Money icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="Money"
  viewBox="0 0 24 24"
>
  <path
    d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon component all icons Search icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="Search"
  viewBox="0 0 24 24"
>
  <path
    d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon component all icons Tune icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="Icon -sizeRegular"
  data-testid="Tune"
  viewBox="0 0 24 24"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
  />
</svg>
`;
